src/interpreter/mod.rs

```diff
@@ -1169,58 +1222,89 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 self.stack[frame].set_local(local, Value::ByVal(val));
                 Ok(())
             }
+            Lvalue::Static(cid) => {
+                let static_val = self.statics.get_mut(&cid).expect("static not cached");
+                assert!(static_val.mutable);
```

Comment:
I think you can do it with a `static mut`. Miri should initialize those and freeze them (because in @eddyb's model, Miri shouldn't even mutate `static mut` after creation), so then you could do:

```rust
static mut X: i32 = 42;
unsafe { X = 123; }
```